docs/source/share.rst

```diff
@@ -0,0 +1,662 @@
+
+1. Create a new Dataset card by opening the `online card creator <https://huggingface.co/datasets/card-creator/>`_, or manually copying the template:
+
+   .. code-block::
+
+        cp ./templates/README.md ./datasets/<your_dataset_name>/README.md
+
+2. Next, you need to generate the structured tags. These help users discover your dataset on the Hub. Create the tags with the `online tagging app <https://huggingface.co/datasets/tagging/>`_, or you can clone and install the `Dataset tagging app <https://github.com/huggingface/datasets-tagging>`_ locally.
+
+3. Select the appropriate tags for your dataset from the dropdown menus, and save the file once you are done.
+
+4. Expand the **Show YAML output aggregating the tags** section on the right, copy the YAML tags, and paste it under the matching section on the online form. Paste the tags into your ``README.md`` file if you manually created your Dataset card.
+
+5. Expand the **Show Markdown Data Fields** section, paste it into the **Data Fields** section under **Data Structure** on the online form (or your local ``README.md``). Modify the descriptions as needed, and briefly describe each of the fields.
+
+6. Fill out the Dataset card to the best of your ability. Refer to the `Dataset Card Creation Guide <https://github.com/huggingface/datasets/blob/master/templates/README_guide.md>`_ for more detailed information about each section of the card. For fields you are unable to complete, you can write **[More Information Needed]**.
+
+7. Once you are done filling out the card with the online form, click the **Export** button to download the Dataset card. Place it in the same folder as your dataset.
+
+Upload
+------
+
+The final step is to upload your dataset! There are two types of datasets based on your sharing workflow: canonical and community datasets. The main differences between the two are highlighted in the table below:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Canonical datasets
+      - Community datasets
+    * - Faster to share, no review process.
+      - Slower to add, needs to be reviewed.
+    * - Data files can be stored on the Hub.
+      - Data files are typically retrieved from the original URLs.
+    * - Identified by a user or organization namespace like `thomwolf/my_dataset` or `huggingface/our_dataset`.
+      - Identified by a root namepsace, need to select a short name that is available.
+    * - Flagged as `unsafe` because the dataset contains executable code.
+      - Flagged as `safe` because the dataset has been reviewed.
+
+.. important::
+
+    The distinction between a canonical and community dataset is based solely on the selected sharing workflow. It does not involve any ranking, decisioning, or opinion regarding the contents of the dataset itself.
+
+Canonical dataset
+^^^^^^^^^^^^^^^^^
+
+To share a canonical dataset:
```

Comment:
TODO: explain the pros/cons of sharing a dataset on github rather than the Hugging Face Hub